plasmapy/analysis/fit_functions.py

```diff
@@ -0,0 +1,1095 @@
+    def __str__(self):
+        return f"f(x) = A exp(alpha x)"
+
+    @property
+    def latex_str(self) -> str:
+        return fr"A \, \exp(\alpha x)"
+
+    def func(self, x, a, alpha):
+        """
+        The fit function, a exponential function.
+
+        .. math::
+
+            f(x) = A \\, e^{\\alpha \\, x}
+
+        where :math:`A` and :math:`\\alpha` are real constants and :math:`x`
+        is the independent variable.
+
+        Parameters
+        ----------
+        x: array_like
+            Independent variable.
+
+        a: float
+            value for the exponential "normalization" constant, :math:`A`
+
+        alpha: float
+            value for the growth constant, :math:`\\alpha`
+
+        Returns
+        -------
+        y: array_like
+            dependent variables corresponding to ``x``
+
+        """
+        x = self._check_x(x)
+        self._check_params(a, alpha)
+
+        return a * np.exp(alpha * x)
+
+    def func_err(self, x, x_err=None, rety=False):
+        """
+        Calculate dependent variable uncertainties :math:`\\delta y` for
+        dependent variables :math:`y=f(x)`.
+
+        .. math::
+
+            \\left( \\frac{\\delta y}{|y|} \\right)^2 =
+                \\left( \\frac{\\delta A}{A} \\right)^2
+                + (x \\, \\delta \\alpha)^2
+                + (\\alpha \\, \\delta x)^2
+
+        """
+        x = self._check_x(x)
+        if x_err is not None:
+            x_err = self._check_x(x_err)
+
+            if x_err.shape == ():
+                pass
+            elif x_err.shape != x.shape:
+                raise ValueError(
+                    f"x_err shape {x_err.shape} must be equal the shape of "
+                    f"x {x.shape}."
+                )
+
+        a, alpha = self.params
+        a_err, alpha_err = self.param_errors
+        y = self.func(x, a, alpha)
+
+        a_term = (a_err / a) ** 2
+        alpha_term = (x * alpha_err) ** 2
+
+        err = a_term + alpha_term
+
+        if x_err is not None:
+            x_term = (alpha * x_err) ** 2
+            err += x_term
+
+        err = np.abs(y) * np.sqrt(err)
+
+        if rety:
+            return err, y
+
+        return err
+
+    func_err.__doc__ += _func_err_doc_core
+
+    def root_solve(self, *args, **kwargs):
+        """
+        The root :math:`f(x_r) = 0` for the fit function. **An exponential has no
+        real roots.**
+
+        Parameters
+        ----------
+        *args
+            Not needed.  This is to ensure signature compatibility with
+            `AbstractFitFunction`.
+
+        *kwargs
+            Not needed.  This is to ensure signature compatibility with
+            `AbstractFitFunction`.
+
+        Returns
+        -------
+        root: float
+            The root value for the given fit :attr:`parameters`.
+
+        err: float
+            The uncertainty in the calculated root for the given fit
+            :attr:`parameters` and :attr:`parameters_err`.
+        """
+
```

Comment:
I have to think about this one.  This will always be a `(np.nan, np.nan)`, so the call will always give a warning.  That just seems weird to me.  Especially since `np.nan` is the legitimate value in this case.

Reply:
I have memories of getting a NaN in the final result of an IDL program, and then spending ~half an hour trying to track down where the NaN was introduced.  I think a warning would make situations like that easier to deal with.  In any case, I'll trust your judgement on this one.